### PR TITLE
use atime_test/atime_test_list to define performance tests

### PR DESCRIPTION
**Pull request by tdhock**
This PR refactors performance tests to reduce repetition.
new atime_test() lets us avoid using quote() in each setup/expr.
new atime_test_list() lets us avoid repeating the same pkg.edit.fun for each performance test.

### CODE REVIEW

File: .ci/atime/tests.R [hunk 1]
```diff
@@ -66,61 +66,60 @@ pkg.edit.fun = function(old.Package, new.Package, sha, new.pkg.path) {
...
+  "setDT improved in #5427" = atime::atime_test(
+    N = 10^seq(1, 7),
+    setup = {
+      L <- replicate(N, 1, simplify = FALSE)
+      setDT(L)
+    },
+    expr = {
+      data.table:::setattr(L, "class", NULL)
+      data.table:::setDT(L)
+    },
```
Comment (tdhock) [inline .ci/atime/tests.R:121]: this is re-indenting to the correct level

File: .ci/atime/tests.R [hunk 2]
```diff
@@ -66,61 +66,60 @@ pkg.edit.fun = function(old.Package, new.Package, sha, new.pkg.path) {
...
+    setup = {
       n <- N/100
       set.seed(2L)
       dt <- data.table(
         g = sample(seq_len(n), N, TRUE),
         x = runif(N),
         key = "g")
       dt_mod <- copy(dt)
-    }),
-    expr = quote(data.table:::`[.data.table`(dt_mod, , N := .N, by = g)),
```
Comment (tdhock) [inline .ci/atime/tests.R:105]: using atime_test, quote is no longer needed in setup, nor expr

File: .ci/atime/tests.R [hunk 3]
```diff
@@ -66,61 +66,60 @@ pkg.edit.fun = function(old.Package, new.Package, sha, new.pkg.path) {
...
+#   When there was no regression, use 'Slow' and 'Fast' 
 # @note Please check https://github.com/tdhock/atime/blob/main/vignettes/data.table.Rmd for more information.
 # nolint start: undesirable_operator_linter. ':::' needed+appropriate here.
-test.list <- list(
+test.list <- atime::atime_test_list(
   # Performance regression discussed in: https://github.com/Rdatatable/data.table/issues/4311
   # Fixed in: https://github.com/Rdatatable/data.table/pull/4440
-  "shallow regression fixed in #4440" = list(
-    pkg.edit.fun = pkg.edit.fun,
+  pkg.edit.fun = pkg.edit.fun,
```
Comment (tdhock) [inline .ci/atime/tests.R:78]: using atime_test_list, pkg.edit.fun now can now be defined once here, and then shared among all tests

File: .ci/atime/tests.R (hunk 2 above)
Comment (Anirban166) [reply .ci/atime/tests.R:105]: Sweet!

**Review by Anirban166 — APPROVED** [review]

File: .ci/atime/tests.R (hunk 3 above)
Comment (Anirban166) [reply .ci/atime/tests.R:78]: Looks like it isn't being detected:
```sh
Error in atime::atime_pkg(Sys.getenv("GITHUB_WORKSPACE"), tests.dir = ".ci") : 
  object 'pkg.edit.fun' not found
Calls: <Anonymous> -> do.call -> <Anonymous>
Execution halted
```
(I tested it on my fork with these changes and got the same too)

Comment (Anirban166) [reply .ci/atime/tests.R:78]: Seems to be an issue with the scope (defining the function inside the `atime::atime_test_list` call fixed it for me on my fork)

**Comment by MichaelChirico** [comment]
Looks great, nice improvement!

**Review by MichaelChirico — APPROVED** [review]

File: .ci/atime/tests.R [hunk 4]
```diff
@@ -73,9 +17,64 @@ pkg.edit.fun = function(old.Package, new.Package, sha, new.pkg.path) {
...
+  # - Makevars, customizing the shared object file name and adjusting the build settings.
+  # - R/onLoad.R, adapting custom version checking for package loading operations.
+  # - NAMESPACE, changing namespace settings for dynamic linking.
+  #
+  # @examples
+  # pkg.edit.fun("data.table", "data.table.some_SHA1_hash", "some_SHA1_hash", "/path/to/data.table")
+  #
+  # @return None (performs in-place file modifications)
+  # @note This setup is typically unnecessary for most packages but essential for data.table due to its unique configuration.
+  pkg.edit.fun = function(old.Package, new.Package, sha, new.pkg.path) {
```
Comment (tdhock) [inline .ci/atime/tests.R:43]: moved pkg.edit.fun inside test.list, and fixed indentation

**Pull request merged**